### PR TITLE
Made a flag `is_strategic` in object.update()

### PR DESCRIPTION
**Pull request by wnykuang**
made `is_strategic` upload flag in object.update()
if `is_stragegic` is true, merge and update object in the original method. 
otherwise, just keep the key/value pairs in the new objects.

make the variable names in `utils` more readable.

### CODE REVIEW

File: pykube/objects.py [hunk 1]
```diff
@@ -12,6 +12,9 @@
 
 
 class ObjectManager:
+    #function all
```
Comment (hjacobs) [inline pykube/objects.py]: these comments don't make sense, Python has docstrings for this

Comment (wnykuang) [reply pykube/objects.py]: indeed, I forgot to clean them up. Sorry about that.

**Comment by wnykuang** [comment]
Made a flag `is_strategic` in object.update() and corresponding utils functions.
default value is True. 

comes with several unittest.

**Comment by hjacobs** [comment]
Thanks!

**Pull request merged**